libbtc won't compile when wallets are disabled

Hello. I discovered an issue recently. With wallets disabled and debug mode enabled, the code won't link.

```
./autogen.sh
./configure --disable-wallet
make
```

 Here's what I'm seeing at the end of the `make` process. This is on Ubuntu 18.04.

```
droark@ubuntu:~/Projects/PersonalGitRepo/libbtc$ make
make[1]: Entering directory '/home/droark/Projects/libbtc'
  CCLD     bitcoin-spv
src/tools/bitcoin_spv-bitcoin-spv.o: In function `main':
/home/droark/libbtc/src/tools/bitcoin-spv.c:166: undefined reference to `btc_wallet_new'
/home/droark/libbtc/src/tools/bitcoin-spv.c:169: undefined reference to `btc_wallet_load'
/home/droark/libbtc/src/tools/bitcoin-spv.c:181: undefined reference to `btc_wallet_set_master_key_copy'
/home/droark/libbtc/src/tools/bitcoin-spv.c:188: undefined reference to `btc_wallet_next_key'
/home/droark/libbtc/src/tools/bitcoin-spv.c:195: undefined reference to `btc_wallet_get_addresses'
/home/droark/libbtc/src/tools/bitcoin-spv.c:204: undefined reference to `btc_wallet_check_transaction'
src/tools/bitcoin_spv-bitcoin-spv.o:/home/droark/Projects/libbtc/./src/logdb/include/logdb/logdb_memdb_llist.h:71: undefined reference to `logdb_llistdb_append'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x8): undefined reference to `logdb_llistdb_init'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x10): undefined reference to `logdb_llist_db_free'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x18): undefined reference to `logdb_llistdb_find'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x20): undefined reference to `logdb_llistdb_size'
src/tools/bitcoin_spv-bitcoin-spv.o:/home/droark/Projects/libbtc/./src/logdb/include/logdb/logdb_memdb_rbtree.h:67: undefined reference to `logdb_rbtree_append'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x48): undefined reference to `logdb_rbtree_init'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x50): undefined reference to `logdb_rbtree_free'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x58): undefined reference to `logdb_rbtree_find'
src/tools/bitcoin_spv-bitcoin-spv.o:(.data.rel+0x60): undefined reference to `logdb_rbtree_size'
./.libs/libbtc.a(libbtc_la-netspv.o):/home/droark/Projects/libbtc/./src/logdb/include/logdb/logdb_memdb_llist.h:71: undefined reference to `logdb_llistdb_append'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x8): undefined reference to `logdb_llistdb_init'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x10): undefined reference to `logdb_llist_db_free'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x18): undefined reference to `logdb_llistdb_find'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x20): undefined reference to `logdb_llistdb_size'
./.libs/libbtc.a(libbtc_la-netspv.o):/home/droark/Projects/libbtc/./src/logdb/include/logdb/logdb_memdb_rbtree.h:67: undefined reference to `logdb_rbtree_append'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x48): undefined reference to `logdb_rbtree_init'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x50): undefined reference to `logdb_rbtree_free'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x58): undefined reference to `logdb_rbtree_find'
./.libs/libbtc.a(libbtc_la-netspv.o):(.data.rel+0x60): undefined reference to `logdb_rbtree_size'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):/home/droark/Projects/libbtc/./src/logdb/include/logdb/logdb_memdb_llist.h:71: undefined reference to `logdb_llistdb_append'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x8): undefined reference to `logdb_llistdb_init'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x10): undefined reference to `logdb_llist_db_free'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x18): undefined reference to `logdb_llistdb_find'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x20): undefined reference to `logdb_llistdb_size'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):/home/droark/Projects/libbtc/./src/logdb/include/logdb/logdb_memdb_rbtree.h:67: undefined reference to `logdb_rbtree_append'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x48): undefined reference to `logdb_rbtree_init'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x50): undefined reference to `logdb_rbtree_free'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x58): undefined reference to `logdb_rbtree_find'
./.libs/libbtc.a(libbtc_la-headersdb_file.o):(.data.rel+0x60): undefined reference to `logdb_rbtree_size'
collect2: error: ld returned 1 exit status
Makefile:1073: recipe for target 'bitcoin-spv' failed
make[1]: *** [bitcoin-spv] Error 1
make[1]: Leaving directory '/home/droark/Projects/libbtc'
Makefile:1899: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

MacOS 10.13.5 with Xcode 9.3.1 generates similar results.

```
droark:libbtc droark$ make
  CCLD     bitcoin-spv
Undefined symbols for architecture x86_64:
  "_btc_wallet_check_transaction", referenced from:
      _main in bitcoin_spv-bitcoin-spv.o
  "_btc_wallet_get_addresses", referenced from:
      _main in bitcoin_spv-bitcoin-spv.o
  "_btc_wallet_load", referenced from:
      _main in bitcoin_spv-bitcoin-spv.o
  "_btc_wallet_new", referenced from:
      _main in bitcoin_spv-bitcoin-spv.o
  "_btc_wallet_next_key", referenced from:
      _main in bitcoin_spv-bitcoin-spv.o
  "_btc_wallet_set_master_key_copy", referenced from:
      _main in bitcoin_spv-bitcoin-spv.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [bitcoin-spv] Error 1
make: *** [all-recursive] Error 1
```

I looked for a bit but needed to get back to another project. My workaround was to leave out `--disable-wallet`. If somebody else can fix this quickly, cool, otherwise I'll take a swing at it when I get a moment.

Thanks.

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,6 +34,9 @@ matrix:
         compiler: gcc
         env: VALGRIND_UNIT_TESTS="yes" RUN_TOOLTESTS="yes" VALGRIND_TOOLTESTS="yes" SUBMIT_COVERALLS="yes" $BUILD_CONFIG=''
       - os: linux
+        compiler: gcc
+        env: $BUILD_CONFIG='--disable-wallet'
+      - os: linux
         compiler: x86_64-w64-mingw32
         env: HOST="x86_64-w64-mingw32" CROSS_COMPILE="yes" RUN_WINE_UNIT_TESTS="yes" $BUILD_CONFIG="--disable-shared --enable-static --disable-net"
       - os: linux

--- a/Makefile.am
+++ b/Makefile.am
@@ -96,8 +96,7 @@ tests_LDFLAGS = -static
 TESTS = tests
 endif
 
-if WITH_WALLET
-
+if WITH_LOGDB
 libbtc_la_SOURCES += \
     src/logdb/logdb_core.c \
     src/logdb/logdb_memdb_llist.c \
@@ -115,6 +114,14 @@ logdbinclude_HEADERS = \
     src/logdb/include/logdb/logdb.h \
     src/logdb/include/logdb/red_black_tree.h
 
+if USE_TESTS
+tests_SOURCES += \
+    src/logdb/test/logdb_tests.c \
+    src/logdb/test/tests_red_black_tree.c
+endif
+endif
+
+if WITH_WALLET
 libbtcinclude_HEADERS += \
     include/btc/wallet.h
 
@@ -123,9 +130,7 @@ libbtc_la_SOURCES += \
 
 if USE_TESTS
 tests_SOURCES += \
-    test/wallet_tests.c \
-    src/logdb/test/logdb_tests.c \
-    src/logdb/test/tests_red_black_tree.c
+    test/wallet_tests.c
 endif
 endif
 

--- a/configure.ac
+++ b/configure.ac
@@ -132,6 +132,7 @@ AM_CONDITIONAL([USE_TESTS], [test x"$use_tests" != x"no"])
 AM_CONDITIONAL([WITH_TOOLS], [test "x$with_tools" = "xyes"])
 AM_CONDITIONAL([WITH_WALLET], [test "x$with_wallet" = "xyes"])
 AM_CONDITIONAL([WITH_NET], [test "x$with_net" = "xyes"])
+AM_CONDITIONAL([WITH_LOGDB], [test "x$with_tools" = "xyes" -o "x$with_wallet" = "xyes"])
 
 ac_configure_args="${ac_configure_args} --enable-module-recovery"
 AC_CONFIG_SUBDIRS([src/secp256k1])

--- a/src/tools/bitcoin-spv.c
+++ b/src/tools/bitcoin-spv.c
@@ -163,6 +163,10 @@ int main(int argc, char* argv[])
 
     if (strcmp(data, "scan") == 0) {
         btc_ecc_start();
+        btc_spv_client* client = btc_spv_client_new(chain, debug, (dbfile && (dbfile[0] == '0' || (strlen(dbfile) > 1 && dbfile[0] == 'n' && dbfile[0] == 'o'))) ? true : false);
+        client->header_message_processed = spv_header_message_processed;
+        client->sync_completed = spv_sync_completed;
+        #if WITH_WALLEt
         btc_wallet *wallet = btc_wallet_new(chain);
         int error;
         btc_bool created;
@@ -202,11 +206,9 @@ int main(int argc, char* argv[])
             printf("Addr: %s\n", addr);
         }
         vector_free(addrs, true);
-        btc_spv_client* client = btc_spv_client_new(chain, debug, (dbfile && (dbfile[0] == '0' || (strlen(dbfile) > 1 && dbfile[0] == 'n' && dbfile[0] == 'o'))) ? true : false);
-        client->header_message_processed = spv_header_message_processed;
-        client->sync_completed = spv_sync_completed;
         client->sync_transaction = btc_wallet_check_transaction;
         client->sync_transaction_ctx = wallet;
+        #endif
         if (!btc_spv_client_load(client, (dbfile ? dbfile : "headers.db"))) {
             printf("Could not load or create headers database...aborting\n");
             ret = EXIT_FAILURE;

